syntax = "proto3";

package com.example.protobuf;

// Order entity for e-commerce system
message Order {
  string order_id = 1;
  string customer_id = 2;
  string order_date = 3;
  string status = 4;  // pending, confirmed, shipped, delivered, cancelled
  
  // Order items
  message OrderItem {
    string product_id = 1;
    int32 quantity = 2;
    double unit_price = 3;
    double discount = 4;
    double subtotal = 5;
  }
  
  repeated OrderItem order_items = 5;
  
  // Financial details
  double subtotal = 6;
  double tax = 7;
  double shipping_cost = 8;
  double total_amount = 9;
  
  // Shipping address
  message ShippingAddress {
    string recipient_name = 1;
    string street = 2;
    string city = 3;
    string postal_code = 4;
    string country_code = 5;
    string phone = 6;
  }
  
  ShippingAddress shipping_address = 10;
  
  // Metadata
  string created_at = 11;
  string updated_at = 12;
}